test/rsht_test.c

```diff
@@ -14,13 +14,21 @@ bool lookerupper(rsht_entry *ent, void *userdata) {
   return true;
 }
 
-
 bool destructor(rsht_entry *ent, void *userdata) {
   free(ent->key);
   free(ent->val);
   return true;
 }
 
+// this will return false on the 5th iteration
+bool failer(rsht_entry *ent, void *userdata)
+{
+  static size_t fail_counter = 5;
```

Comment:
I'm not sure I love this variable being in here, but I guess it makes sense. The comment was really helpful in understanding it, so I think it's okay.